service/src/main/java/tds/assessment/repositories/impl/AssessmentQueryRepositoryImpl.java

```diff
@@ -182,17 +186,42 @@ public AssessmentQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTempla
                 "GROUP BY \n" +
                 "   assessmentKey, assessmentId, subject, assessmentLabel, maxAttempts";
 
-        return jdbcTemplate.query(SQL, parameters, (rs, row) ->
-            new AssessmentInfo.Builder()
-                .withKey(rs.getString("assessmentKey"))
-                .withId(rs.getString("assessmentId"))
-                .withLabel(rs.getString("assessmentLabel"))
-                .withSubject(rs.getString("subject"))
-                .withMaxAttempts(rs.getInt("maxAttempts"))
-                .withLanguages(Arrays.asList(rs.getString("languages").split(",")))
-                .withGrades(COMMA_SPLITTER.splitToList(rs.getString("grades")))
-                .build()
-        );
+        return jdbcTemplate.query(SQL, parameters, new AssessmentInfoRowMapper());
+    }
+
+    @Override
+    public List<AssessmentInfo> findAssessmentInfoForGrade(final String clientName, final String grade) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("grade", grade)
+            .addValue("clientName", clientName);
+
+        final String SQL =
+            "SELECT \n" +
```

Comment:
Suggest pull this select, from, and joins into a constant method that the two methods share.  That way if we need to change the mapper for whatever reason we can change the query in one place and the mapper.

At least it looks like the only difference is the where and group by.